File: 1168-duplicate-zeros/1168-duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        length = len(arr)
        idx = 0

        while idx < length:
            if arr[idx] == 0:
                arr.insert(idx + 1,0)
                arr.pop()
                idx += 1
            idx += 1
                
            

File: 1168-duplicate-zeros/1168-duplicate-zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    len = arr.length
    index = 0

    while(index < len){
        if(arr[index] === 0){
            arr.splice(index + 1,0,0)
            arr.pop()
            index++
        }
        index++
    }
};